### PR TITLE
Select company on login page

### PR DESCRIPTION
**Pull request by bafbes**
As in Dolibarr, when multicompany is active, a select input is shown in login page so as to login in the right one

### CODE REVIEW

File: www/lib/login.lib.php [hunk 1]
```diff
@@ -45,6 +45,7 @@ function dol_loginfunction($langs,$conf,$mysoc)
     }
 
     // Execute hook getLoginPageOptions (for table)
+    $hookmanager->initHooks(array('mainloginpage'));
```
Comment (atm-john) [inline www/lib/login.lib.php:48]:
```suggestion
    $hookmanager->initHooks(array('externalaccessloginpage'));
```

**Review by atm-john — CHANGES_REQUESTED** [review]
Thanks @bafbes good to see you ;-)

Your pull request is interesting but I can't merge it because we dont't want to display all entity on external access login for privacy / security reasons and to keep login simple for external users.
Adding hook here is a good idea, but need to change hookname, to avoid incompatible module to add incompatible stuff...

But you can use this conf EACCESS_FORCE_ENTITY_ON_LOGIN

We're currently looking for a solution to handle multicompany cases, allowing us to manage entities by mapping domain name to enties because login page need to reflect logo and style according to entitie.

File: www/lib/login.lib.php (hunk 1 above)
Comment (bafbes) [reply www/lib/login.lib.php:48]: Hi John, I am glad to see you too.
I just mad a pull request with the devs I made for a customer. I let it ready for any other future development decision as there are so many alternatives : We may set an option to enable this function or select a single portal entity. We may also provide an url parameter to do so.

Comment (atm-john) [reply www/lib/login.lib.php:48]: if you change the name of hook i can merge.

**Comment by bafbes** [comment]
Done.

File: www/tpl/form.login.tpl.php [hunk 2]
```diff
@@ -64,7 +64,13 @@
...
                     </label>
                 </div>
-                <?php } ?>
+                <?php }
+                if (!empty($morelogincontent) && !is_array($morelogincontent)) {
+                    $morelogincontent = "<span class='form-control'>$morelogincontent</span><br>";
+                    echo '<!-- Option by hook -->';
+                    echo $morelogincontent;
+                }
+                ?>
```
Comment (atm-john) [inline www/tpl/form.login.tpl.php:73]:
```suggestion
                <?php }
                if (!empty($morelogincontent) && !is_array($morelogincontent)) {
                    print '<!-- Start option by hook getLoginPageOptions -->''.$morelogincontent.'<!-- End option by hook getLoginPageOptions -->'';
                }
                ?>
```

**Pull request merged**